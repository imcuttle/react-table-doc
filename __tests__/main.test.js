/**
 * @file main
 * @author imcuttle
 * @date 2018/4/4
 */
const main = require('../')
const { parse } = require('react-docgen')
const { readFileSync, writeFileSync } = require('fs')
const { makeFixture } = require('./helper')

describe('main', function() {
  beforeAll(function () {
    writeFileSync(
      makeFixture('button.json'),
      JSON.stringify(
        parse(readFileSync(makeFixture('Button.js')).toString()),
        null,
        2
      )
    )
  })

  it('should react-docgen passed', function() {
    expect(require(makeFixture('button.json'))).toMatchSnapshot()
  })
})
