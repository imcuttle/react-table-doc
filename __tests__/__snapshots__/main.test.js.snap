// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main should react-docgen passed 1`] = `
Object {
  "description": "是多少 哈哈哈哈
### 666
- test",
  "displayName": "Button",
  "methods": Array [
    Object {
      "docblock": null,
      "modifiers": Array [],
      "name": "handleClick",
      "params": Array [
        Object {
          "name": "e",
          "type": null,
        },
      ],
      "returns": null,
    },
    Object {
      "docblock": null,
      "modifiers": Array [],
      "name": "updateLoadingPosition",
      "params": Array [],
      "returns": null,
    },
    Object {
      "docblock": null,
      "modifiers": Array [],
      "name": "setAsyncLoading",
      "params": Array [
        Object {
          "name": "loading",
          "type": null,
        },
      ],
      "returns": null,
    },
    Object {
      "docblock": null,
      "modifiers": Array [],
      "name": "setShowLoading",
      "params": Array [
        Object {
          "name": "showLoading",
          "type": null,
        },
      ],
      "returns": null,
    },
    Object {
      "docblock": null,
      "modifiers": Array [],
      "name": "compClass",
      "params": Array [],
      "returns": null,
    },
    Object {
      "docblock": null,
      "modifiers": Array [],
      "name": "loadingIconName",
      "params": Array [],
      "returns": null,
    },
  ],
  "props": Object {
    "classPrefix": Object {
      "defaultValue": Object {
        "computed": false,
        "value": "'erp'",
      },
      "description": "",
      "required": false,
      "type": Object {
        "name": "string",
      },
    },
    "disabled": Object {
      "description": "",
      "required": false,
      "type": Object {
        "name": "bool",
      },
    },
    "icon": Object {
      "defaultValue": Object {
        "computed": false,
        "value": "''",
      },
      "description": "",
      "required": false,
      "type": Object {
        "name": "string",
      },
    },
    "iconSize": Object {
      "defaultValue": Object {
        "computed": false,
        "value": "'normal'",
      },
      "required": false,
    },
    "loading": Object {
      "description": "",
      "required": false,
      "type": Object {
        "name": "bool",
      },
    },
    "loadingSpinDelayInMS": Object {
      "defaultValue": Object {
        "computed": false,
        "value": "300",
      },
      "description": "",
      "required": false,
      "type": Object {
        "name": "number",
      },
    },
    "onClick": Object {
      "description": "",
      "required": false,
      "type": Object {
        "name": "func",
      },
    },
    "onClickDone": Object {
      "description": "",
      "required": false,
      "type": Object {
        "name": "func",
      },
    },
    "size": Object {
      "defaultValue": Object {
        "computed": false,
        "value": "'default'",
      },
      "description": "尺寸 'default' | 'small' | 'x-small' | 'large'
实际没有large，因为规范上没有定义
default: 'default'",
      "required": false,
      "type": Object {
        "name": "enum",
        "value": Array [
          Object {
            "computed": false,
            "value": "'default'",
          },
          Object {
            "computed": false,
            "value": "'small'",
          },
          Object {
            "computed": false,
            "value": "'large'",
          },
          Object {
            "computed": false,
            "value": "'x-small'",
          },
        ],
      },
    },
    "type": Object {
      "defaultValue": Object {
        "computed": false,
        "value": "'default'",
      },
      "description": "按钮类型类型 'default' | 'primary' | 'success' | 'danger' | 'normal' | 'pale'
default: 'default'",
      "required": false,
      "type": Object {
        "name": "enum",
        "value": Array [
          Object {
            "computed": false,
            "value": "'default'",
          },
          Object {
            "computed": false,
            "value": "'primary'",
          },
          Object {
            "computed": false,
            "value": "'normal'",
          },
          Object {
            "computed": false,
            "value": "'success'",
          },
          Object {
            "computed": false,
            "value": "'danger'",
          },
          Object {
            "computed": false,
            "value": "'pale'",
          },
          Object {
            "computed": false,
            "value": "'bare'",
          },
          Object {
            "computed": false,
            "value": "'link'",
          },
        ],
      },
    },
  },
}
`;
